packages/aws-cdk-lib/aws-sns/lib/topic-base.ts

```diff
@@ -122,21 +136,43 @@ export abstract class TopicBase extends Resource implements ITopic {
    * Adds a statement to the IAM resource policy associated with this topic.
    *
    * If this topic was created in this stack (`new Topic`), a topic policy
-   * will be automatically created upon the first call to `addToPolicy`. If
+   * will be automatically created upon the first call to `addToResourcePolicy`. If
    * the topic is imported (`Topic.import`), then this is a no-op.
    */
-  public addToResourcePolicy(statement: iam.PolicyStatement): iam.AddToResourcePolicyResult {
+  public addToResourcePolicy(statement: iam.PolicyStatement, addToResourcePolicyProps?: AddToResourcePolicyProps): iam.AddToResourcePolicyResult {
```

Comment:
I don't think adding a new function argument here is the right way to do this. The `addToResourcePolicy(statement: iam.PolicyStatement)` is a pattern used throughout the code base. I think a better solution is to just make `enforceSSL` a protected attribute of the base class:
`protected enforceSSL?: boolean`. We can then add `enforceSSL` as a property on `TopicProps` and set it in the constructor. So:

```ts
export interface TopicProps {
	// ...
	enforceSSL?: boolean;
}

export class Topic extends TopicBase {
	// ...
	constructor(scope: Construct, id: string, props: TopicProps = {}) {
		// ...
		this.enforceSSL = props.enforceSSL;
	}
}
```

Then in `addToResourcePolicy` we just check `this.enforceSSL` and reuse the logic you've added with `createSSLPolicyDocument`.

Thoughts?